fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/Skolemizer.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.kernel.impl.rdf;
+
+import static com.google.common.hash.Hashing.murmur3_32;
+import static com.hp.hpl.jena.rdf.model.ResourceFactory.createResource;
+import static com.hp.hpl.jena.rdf.model.ResourceFactory.createStatement;
+import static java.util.UUID.randomUUID;
+import static org.fcrepo.kernel.RdfLexicon.REPOSITORY_NAMESPACE;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.google.common.hash.HashCode;
+import com.hp.hpl.jena.rdf.model.AnonId;
+import com.hp.hpl.jena.rdf.model.Literal;
+import com.hp.hpl.jena.rdf.model.Property;
+import com.hp.hpl.jena.rdf.model.RDFNode;
+import com.hp.hpl.jena.rdf.model.RDFVisitor;
+import com.hp.hpl.jena.rdf.model.Resource;
+import com.hp.hpl.jena.rdf.model.Statement;
+
+/**
+ * Skolemization is abstractly a function from RDF nodes to RDF nodes, but here we implement it, purely for
+ * convenience of operation, as a function from triples to triples. {@link #nodeSkolemizer} represents the
+ * node-to-node function. This class will make two blank nodes with the same identifier into the same skolem resource
+ * because it uses only one prefix for its lifetime and because the mapping from a given blank node to a skolem
+ * resource depends only on that prefix and on a MurmurHash3 of the identifier of the blank node in question. An
+ * instance of this class should be used only with a contextual topic of one single resource and only for one
+ * document's scope of RDF about that resource.
+ *
+ * @author ajs6f
+ */
+public class Skolemizer implements Function<Statement, Statement>, Supplier<Set<Resource>> {
+
+    private final RDFVisitor nodeSkolemizer;
+
+    public static final Resource SKOLEM_TYPE = createResource(REPOSITORY_NAMESPACE + "skolem");
```

Comment:
I think you're looking at a previous commit. That's not there anymore.

---
A. Soroka
The University of Virginia Library

On Mar 18, 2015, at 4:19 PM, Andrew Woods <notifications@github.com> wrote:

> In fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/Skolemizer.java:
> 
> > +/**
> > + * Skolemization is abstractly a function from RDF nodes to RDF nodes, but here we implement it, purely for
> > + * convenience of operation, as a function from triples to triples. {@link #nodeSkolemizer} represents the
> > + * node-to-node function. This class will make two blank nodes with the same identifier into the same skolem resource
> > + * because it uses only one prefix for its lifetime and because the mapping from a given blank node to a skolem
> > + * resource depends only on that prefix and on a MurmurHash3 of the identifier of the blank node in question. An
> > + * instance of this class should be used only with a contextual topic of one single resource and only for one
> > + * document's scope of RDF about that resource.
> > + *
> > + * @author ajs6f
> > + */
> > +public class Skolemizer implements Function<Statement, Statement>, Supplier<Set<Resource>> {
> > +
> > +    private final RDFVisitor nodeSkolemizer;
> > +
> > +    public static final Resource SKOLEM_TYPE = createResource(REPOSITORY_NAMESPACE + "skolem");
> 
> Remove unused constant SKOLEM_TYPE.
> 
> �
> Reply to this email directly or view it on GitHub.
>